heron/common/src/cpp/network/baseconnection.cpp

```diff
@@ -50,6 +53,67 @@ BaseConnection::BaseConnection(ConnectionEndPoint* endpoint, ConnectionOptions*
   mState = INIT;
   mOnClose = NULL;
   bufferevent_options boptions = BEV_OPT_DEFER_CALLBACKS;
+
+  if (mOptions->ssloptions_.is_configured()) {
+    LOG(INFO) << "SSL is configured with certificate path"
+              << mOptions->ssloptions_.get_certificate_path() << " and private key path "
+              << mOptions->ssloptions_.get_private_key_path();
+    openSSLConnection(boptions);
+  } else {
+    openConnection(boptions);
+  }
+}
+
+void BaseConnection::openSSLConnection(bufferevent_options boptions) {
+  LOG(INFO) << "Starting connection with SSL cert path " <<
+    mOptions->ssloptions_.get_certificate_path() << " and key path " <<
+    mOptions->ssloptions_.get_private_key_path();
+
+  SSL_CTX *ctx;
+  if ((ctx = SSL_CTX_new(SSLv23_method())) == NULL) {
```

Comment:
Is there a corresponding call to `SSL_CTX_free()` somewhere? If not, this is potentially a leak